benchmarks/polys.py

```diff
@@ -1,17 +1,148 @@
+        D = 1 + x ** (n + 1) + sum([y[i] ** (n + 1) for i in range(n)])
+        f = D * (-2 + x ** (n + 1) + sum([y[i] ** (n + 1) for i in range(n)]))
+        g = D * (2 + x ** (n + 1) + sum([y[i] ** (n + 1) for i in range(n)]))
+        fp, gp = Poly(f, x, *y[:n]), Poly(g, x, *y[:n])
+        fpe, gpe = R.from_sympy(f), R.from_sympy(g)
+        return f, g, fp, gp, fpe, gpe, x, y, R
+
+
+class _QuadraticNonMonicGCD:
+    def generate(self, n):
+        self.x, *self.y = symbols("x, y1:9")
+        self.R1 = [self.x] + list(self.y)
+        self.R = ZZ[self.R1]
+        x, y, R = self.x, self.y, self.R
+        D = 1 + x ** 2 * y[0] ** 2 + sum([y[i] ** 2 for i in range(1, n)])
+        f = D * (-1 + x ** 2 - y[0] ** 2 + sum([y[i] ** 2 for i in range(1, n)]))
+        g = D * (2 + x * y[0] + sum(y[1:n])) ** 2
+        fp, gp = Poly(f, x, *y[:n]), Poly(g, x, *y[:n])
+        fpe, gpe = R.from_sympy(f), R.from_sympy(g)
+        return f, g, fp, gp, fpe, gpe, x, y, R
+
+
+class _SparseNonMonicQuadratic:
+    def generate(self, n):
+        self.x, *self.y = symbols("x, y1:9")
+        self.R1 = [self.x] + list(self.y)
+        self.R = ZZ[self.R1]
+        x, y, R = self.x, self.y, self.R
+        D = -1 + x * prod(y[:n])
+        f = D * (3 + x * prod(y[:n]))
+        g = D * (-3 + x * prod(y[:n]))
+        fp, gp = Poly(f, x, *y[:n]), Poly(g, x, *y[:n])
+        fpe, gpe = R.from_sympy(f), R.from_sympy(g)
+        return f, g, fp, gp, fpe, gpe, x, y, R
+
+
+class _TimePREM:
+    """Benchmarks for `prem` method in polynomials."""
+
+    def setup_polys(self, n):
+        (f, g, fp, gp, fpe, gpe, x, y, R) = self.generate(n)
+
+        self.values = {}
+        self.f = f
+        self.g = g
+        self.fp = fp
+        self.gp = gp
+        self.fpe = fpe
+        self.gpe = gpe
+        self.x = x
+        self.y = y
+        self.R = R
+
+    def setup(self, method, n):
+        self.setup_polys(n)
+
+        f, g, x, R = self.f, self.g, self.x, self.R
+        prem_f_g_x = rem(f * LC(g, x) ** (degree(f, x) - degree(g, x) + 1), g, x)
+
+        self.ref_expr = prem_f_g_x
+        self.ref_ring = R(prem_f_g_x.as_expr())
+
+        if method == 'prem':
+            self.func = lambda: prem(self.f, self.g, self.x)
+
+        elif method == 'poly':
+            self.func = lambda: self.fp.prem(self.gp)
+
+        elif method == 'sparse':
+            self.func = lambda: self.fpe.prem(self.gpe)
+
+    def teardown(self, method, n):
+        for key, val in self.values.items():
+            if key == 'sparse':
+                if (self.ref_ring - val) != 0:
+                    raise ValueError("Incorrect result, invalid timing:"
+                                        " %s != %s" % (self.ref_ring, val))
+
+            elif (self.ref_expr - val).simplify() != 0:
+                raise ValueError("Incorrect result, invalid timing:"
+                                    " %s != %s" % (self.ref_expr, val))
+
+    def time_prem_methods(self, method, n):
```

Comment:
if I changed the order like this `time_(self, n, method)` and 
`params = [(1, 3 , 5, 8), ('prem', 'poly', 'sparse')]`
 then the asv result will looks like this 
 
 ```
[100.00%] ··· polys.TimePREMSparseNonMonicQuadratic.time_prem_methods                                                                                                                                      ok
[100.00%] ··· ======== ============ ============= ============
              --                        param2
              -------- ---------------------------------------
               param1      prem          poly        sparse
              ======== ============ ============= ============
                 1      850±500μs      48.4±2μs    2.67±0.3ms
                 3      581±100μs    1.46±0.03ms   4.73±0.3ms
                 5       536±20μs     4.09±0.2ms   6.89±0.3ms
                 8      1.11±0.6ms    8.14±0.5ms   7.97±0.4ms
              ======== ============ ============= ============
```
I think the above result would not have any problem with the terminal width.

Reply:
Yes, that's good. The only thing that would make it better is to set the `param_names` so instead of `param1/2` it could be `degree`  and `implementation` or something like that.